packages/react-native-gesture-handler/apple/RNGestureHandler.mm

```diff
@@ -249,12 +266,27 @@ - (RNGestureHandlerEventExtraData *)eventExtraData:(UIGestureRecognizer *)recogn
 #endif
 }
 
+/**
+ This method is used in `handleGesture` to choose appropriate view. `reactTag` in `RCTParagraphComponentView`
+ is `nil`, therefore we want to use `reactTag` from `RCTParagraphTextView`.
+ */
+- (RNGHUIView *)chooseViewForInteraction:(UIGestureRecognizer *)recognizer
+{
+#if RCT_NEW_ARCH_ENABLED
+  return [self isViewParagraphComponent:recognizer.view] ? recognizer.view.subviews[0] : recognizer.view;
+#else
+  return recognizer.view;
+#endif
+}
+
 - (void)handleGesture:(UIGestureRecognizer *)recognizer
 {
+  RNGHUIView *view = [self chooseViewForInteraction:recognizer];
+
   // it may happen that the gesture recognizer is reset after it's been unbound from the view,
   // it that recognizer tried to send event, the app would crash because the target of the event
   // would be nil.
-  if (recognizer.view.reactTag == nil) {
+  if (view == nil) {
```

Comment:
```suggestion
  if (view.reactTag == nil) {
```

Are you sure that all other views will have the react tag set?

Reply:
Changed in [36e3441](https://github.com/software-mansion/react-native-gesture-handler/pull/3591/commits/36e3441b92174277f806f3f2fbea28b7f13bf730)